2641 incorrect module names in psydata

hiker opened the pull request:
Fixes #2641.

arporter commented on src/psyclone/psyir/nodes/psy_data_node.py:
```diff
@@ -756,10 +757,16 @@ def gen_type_bound_call(typename, methodname, argument_list=None,
         if module_name is None:
-            module_name = routine_schedule.name
+            container = routine_schedule.ancestor(Container)
+            # See if there current code is inside of a module. If so,
```
"If the current code is inside a module use the module name, otherwise ...."

hiker replied:
Done.

arporter reviewed (changes requested):
Looks good Joerg. Thanks for tracking this down.
You'll need to update the tutorials too - at least `tutorial/practicals/nemo/2_nemo_profiling` needs tweaking.
Does the User/Dev Guide have examples that need updating too?

hiker commented on examples/gocean/eg5/profile/test.x90:
```diff
@@ -37,10 +37,10 @@
-!>  PostEnd called for module 'psy_test' region 'invoke_1_update_field:update_field_code:r0'
+!>  PreStart called for module 'psy_test' region 'invoke_0-r0'
+!>  PostEnd called for module 'psy_test' region 'invoke_0-r0'
+!>  PreStart called for module 'psy_test' region 'invoke_1_update_field:update_field_code-r0'
```
They should indeed be.

hiker commented:
> Looks good Joerg. Thanks for tracking this down. You'll need to update the tutorials too - at least `tutorial/practicals/nemo/2_nemo_profiling` needs tweaking. 

Done, indeed I missed the example in the Fortran file (maybe I should just have removed that :) ).

> Does the User/Dev Guide have examples that need updating too?

Yes, it did - a bit strange, I was certain I had updated that section, but maybe I am mixing things up. I've updated the user guide, added examples for names used in transformation, and shortened the example for DSL naming (and properly indented them).

arporter reviewed (approved):
All requested changes have been made, thanks. (I too got a sense of deja vu when looking at the UG changes but I think that's because your last PR also changed things there?)
Updated UG builds fine and looks good.
Will proceed to merge.
